app/src/test/java/org/dhis2/usecases/main/MainPresenterTest.kt

```diff
@@ -0,0 +1,168 @@
+package org.dhis2.usecases.main
+
+import androidx.work.WorkManager
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.reactivex.Completable
+import io.reactivex.Single
+import org.dhis2.data.prefs.Preference.Companion.DEFAULT_CAT_COMBO
+import org.dhis2.data.prefs.Preference.Companion.PIN
+import org.dhis2.data.prefs.Preference.Companion.PREF_DEFAULT_CAT_OPTION_COMBO
+import org.dhis2.data.prefs.Preference.Companion.SESSION_LOCKED
+import org.dhis2.data.prefs.PreferenceProvider
+import org.dhis2.data.schedulers.SchedulerProvider
+import org.dhis2.data.schedulers.TrampolineSchedulerProvider
+import org.dhis2.usescases.login.LoginActivity
+import org.dhis2.usescases.main.MainPresenter
+import org.dhis2.usescases.main.MainView
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.category.CategoryCombo
+import org.hisp.dhis.android.core.category.CategoryOptionCombo
+import org.hisp.dhis.android.core.user.User
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+/**
+ * Created by frodriguez on 10/22/2019.
+ *
+ */
+class MainPresenterTest {
+
+    private lateinit var presenter: MainPresenter
+    private val schedulers: SchedulerProvider = TrampolineSchedulerProvider()
+    private val view: MainView = mock()
+    private val d2: D2 = mock()
+    private val preferences: PreferenceProvider = mock()
+    private val workManger: WorkManager = mock()
+
+    @Before
+    fun setUp() {
+        presenter = MainPresenter(view, d2, schedulers, preferences, workManger)
+    }
+
+    @Test
+    fun `Should show username on activity resumed`() {
```

Comment:
These tests belong to the same **flow**, function and usecase which is init() the presener. Perhaps they should be compacted as one test... Let me know what you think :+1: 

`presenterMocks()
presenter.init()
//all verifies`